astropy/uncertainty/core.py

```diff
@@ -276,11 +276,17 @@ class ArrayDistribution(Distribution, np.ndarray):
     def view(self, dtype=None, type=None):
         if type is None:
             if issubclass(dtype, np.ndarray):
-                type = dtype
-                dtype = None
+                if issubclass(dtype, Distribution):
+                    return super().view(dtype)
+                else:
+                    result = self.distribution.view(dtype)
+                    return Distribution(result)
```

Comment:
Nitpick and feel free to ignore. I'm just thinking of McCabe complexity. I endeavor to structure code to minimize the complexity score and reducing the number of return statements is an easy couple points.

Reply:
Numpy's strange choice to allow one to omit `dtype` and just pass in a class remains a bit annoying... I've made it slightly simpler (and found a bug on the way, so good!).